src/frontend/utils/conversejs/converse-plugins/marshaJoinDiscussionPlugin.ts

```diff
@@ -0,0 +1,268 @@
+import { appData, getDecodedJwt } from 'data/appData';
+import { useJoinParticipant } from 'data/stores/useJoinParticipant';
+import { useParticipantWorkflow } from 'data/stores/useParticipantWorkflow';
+import { useVideo } from 'data/stores/useVideo';
+import { Participant } from 'types/Participant';
+import { Video } from 'types/tracks';
+import { EventType, MessageType, XMPP } from 'types/XMPP';
+import { converse } from 'utils/window';
+
+const PLUGIN_NAME = 'marsha-join-discussion-plugin';
+
+const addMarshaJoinDiscussionPlugin = (xmpp: XMPP) =>
+  converse.plugins.add(PLUGIN_NAME, {
+    dependencies: ['converse-muc'],
+    initialize() {
+      const _converse = this._converse;
+      let joinedRoom = false;
+      _converse.on('chatRoomInitialized', (model: any) => {
+        model.session.on('change:connection_status', (currentSession: any) => {
+          if (
+            currentSession.get('connection_status') ===
+            converse.ROOMSTATUS.ENTERED
+          ) {
+            joinedRoom = true;
+          }
+          if (
+            currentSession.get('connection_status') ===
+            converse.ROOMSTATUS.DISCONNECTED
+          ) {
+            joinedRoom = false;
+          }
+
+          if (
+            currentSession.get('connection_status') ===
+              converse.ROOMSTATUS.NICKNAME_REQUIRED &&
+            useParticipantWorkflow.getState().asked
+          ) {
+            useParticipantWorkflow.getState().setUsernameAlreadyExisting();
+          }
+        });
+      });
+
+      _converse.on('enteredNewRoom', () => {
+        joinedRoom = true;
+        if (useParticipantWorkflow.getState().asked) {
+          askParticipantToJoin();
+        }
+      });
+
+      _converse.on('initialized', () => {
+        _converse.connection.addHandler(
+          (messageStanza: any) => {
+            if (
+              getDecodedJwt().permissions.can_update &&
```

Comment:
getDecodedJwt().permissions.can_update can be store in a const and used in the rest of the code

canUpdate = getDecodedJwt().permissions.can_update

if (canUpdate ...
else if canUpdate...